llm4pddl/utils.py

```diff
@@ -176,6 +176,33 @@ def get_custom_task(benchmark_name: str, task_num: int) -> Task:
     return Task(domain_file, problem_file)
 
 
+def prep_problem(problem: str) -> str:
+    """Maps a string of a problem file to a new string.
+
+    This new string has had all space removed between grounded atoms.
+    This is for use before querying llm in order to reduce tokens.
+    """
+    # Gets rid of leading and trailing spaces in every line:
+    new = [piece.strip() for piece in problem.split('\n')]
+    # Moves ')' characters to previous line.
+    new.reverse()
+    for i in range(len(new)):
+        piece = new[i]
+        while len(piece) > 0:
+            if piece[0] == ')':
+                new[i] = piece[1:]
+                piece = piece[1:]
+                new[i + 1] = new[i + 1] + ')'
+            else:
+                print('entered!!!!!!!!!!!!')
+                break
+    new.reverse()
+    # Removes lines with empty space
```

Comment:
you might want to do this before the `replace` thing that I suggested, to handle cases where there are multiple new lines before a right parentheses